src/test/java/eu/elixir/ega/ebi/it/filedatabase/FileControllerIntgTest.java

```diff
@@ -0,0 +1,110 @@
+    }
+    
+    /**
+     * Verify the api call /file/{fileId} and check status is {@link org.apache.http.HttpStatus#SC_OK}.
+     * Also passing wrong Id 000 and expecting the response body to be empty array i.e, []
+     */
+    @Test
+    public void testGetFileForZeroId() {
+        final Response response = REQUEST.get(FILE_CONTROLLER + ID_ZERO);
+        JSONArray jsonArray = new JSONArray(response.body().asString()); 
+        
+        response.then().assertThat().statusCode(SC_OK);
+        assertThat(jsonArray.length(), equalTo(0));
+    }
+    
+    
+    /**
+     * Verify the api call /file/{fileId}/datasets and check status is {@link org.apache.http.HttpStatus#SC_OK}.
+     * Also checks the response fileId & datasetId.
+     */
+    @Test
+    public void testGetFileDataset() {
+        final Response response = REQUEST.get(FILE_CONTROLLER + fileId + "/datasets");
+        JSONArray jsonArray = new JSONArray(response.body().asString()); 
+        JSONObject jsonObject = jsonArray.getJSONObject(0);
+        
+        response.then().assertThat().statusCode(SC_OK);
+        assertThat(jsonArray.length(), equalTo(1));
+        assertThat(jsonObject.getString("fileId"), equalTo(fileId));
+        assertThat(jsonObject.getString("datasetId"), equalTo(datasetId));
+    }
+    
+    /**
+     * Verify the api call /file/{fileId}/datasets and check status is {@link org.apache.http.HttpStatus#SC_OK}.
+     * Also passing wrong Id 000 and expecting the response body to be empty array i.e, []
+     */
+    @Test
+    public void testGetFileDatasetForZeroId() {
+        final Response response = REQUEST.get(FILE_CONTROLLER + ID_ZERO + "/datasets");
+        JSONArray jsonArray = new JSONArray(response.body().asString()); 
+        
+        response.then().assertThat().statusCode(SC_OK);
+        assertThat(jsonArray.length(), equalTo(0));
+    }
+    
+    /**
+     * Verify the api call /file/{fileId}/index and check status is {@link org.apache.http.HttpStatus#SC_OK}.
+     * Also checks the response fileId & indexFileId.
+     */
+    @Test
+    public void testGetFileIndex() {
+        final Response response = REQUEST.get(FILE_CONTROLLER + fileId + "/index");
+        JSONArray jsonArray = new JSONArray(response.body().asString()); 
+        JSONObject jsonObject = jsonArray.getJSONObject(0);
+        
+        response.then().assertThat().statusCode(SC_OK);
+        assertThat(jsonArray.length(), equalTo(1));
+        assertThat(jsonObject.getString("fileId"), equalTo(fileId));
+        assertThat(jsonObject.getString("indexFileId"), equalTo(indexId));
+    }
+    
+    /**
+     * Verify the api call /file/{fileId}/index and check status is {@link org.apache.http.HttpStatus#SC_OK}.
+     * Also passing wrong Id 000 and expecting the response body to be empty array i.e, []
+     */
+    @Test
+    public void testGetFileIndexForZeroId() {
+        final Response response = REQUEST.get(FILE_CONTROLLER + ID_ZERO + "/index");

```

Comment:
The same comment as for the non existing dataset

Reply:
Yes, currently it is implemented like this. It returns empty result array if id not found in the database.